Automattic/sensei
# Add learner courses frontend styles

Some solutions here are intermediary at best. The CSS rules are a mixture of the naming style used for blocks, and the frontend classes in the shortcode/templates that can't be changed easily for now. 

### Changes proposed in this Pull Request

* Update frontend styles for Learner Courses block
* Apply customizations with CSS variables
* Do not load Sensei's `frontend.css` for pages with this block
* Change the block layout to what's currently feasible on the frontend
  * Featured image is a floated 100x100 square, even in grid mode. — there is more nuance in Sensei around these images, including filters and settings. It can also pick up a lesson's image, didn't want to move away from that now. Also, I didn't way an easy way to display these images responsively as per the original grid design.
* Add View Results button in the editor. This button (and View Certificate, and possibly some others) are still added on the frontend as they were before.
* Update color customization for consistency: Primary color affects title, buttons(?) and links in the tabs. Accent color affects the progress bar & completed badge.

Wider changes:
* Add having Sensei blocks on the page for `is_sensei`'s condition — adds `.sensei` class to body on pages using Sensei blocks. 

### Testing instructions

* Add a Learner Courses block to a page, publish & view
* Check that it picks up block styles
* Check that list/grid layout works
* Check that color & progress bar customizations work 
* Check in different themes
* Check that the original My courses (sensei_user_courses) shortcode works as previously
* Check that hiding elements of the course entry (featured image, category, description) works


### Screenshot / Video

<img width="850" alt="image" src="https://user-images.githubusercontent.com/176949/114201028-15635280-9956-11eb-89da-8e37f8f2a58f.png">


<img width="649" alt="image" src="https://user-images.githubusercontent.com/176949/114200521-93732980-9955-11eb-80aa-8295654c55ec.png">
